### PR TITLE
[DOC] rework examples/02_decoding/plot_oasis_vbm

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Relates to #6265

Use of AI tools:
- [ ] LLM tools were used to generate at least part of the content of this pull-request.
- [x] No LLM tools were used to generate any part of the content of this pull-request.

### CODE REVIEW

File: examples/02_decoding/plot_oasis_vbm.py [hunk 1]
```diff
@@ -5,226 +5,248 @@
...
+    For more information
+    see the :ref:`dataset description <oasis_maps>`.
 
+Predictive modeling analysis: VBM bio-markers of aging?
+-------------------------------------------------------
 
-Also, parameters such as the smoothing should be applied to the data and the
-number of features selected by the :term:`ANOVA` step should be set by nested
-cross-validation, as they impact significantly the prediction score.
+We run a standard SVM-ANOVA nilearn pipeline to predict age from the VBM data.
```
Comment (emdupre) [inline examples/02_decoding/plot_oasis_vbm.py]:
```suggestion
We run a Support Vector Regression (SVR) and  :term:`ANOVA` using the nilearn `DecoderRegressor` to predict age from the VBM data.
```

**Review by bthirion — APPROVED** [review]

File: examples/02_decoding/plot_oasis_vbm.py [hunk 2]
```diff
@@ -5,228 +5,251 @@
 This example uses Voxel-Based Morphometry (:term:`VBM`)
 to study the relationship between aging and gray matter density.
 
-The data come from the `OASIS <https://sites.wustl.edu/oasisbrains/>`_ project.
+The data come from the Open Access Series of Imaging Studies
```
Comment (hndgzkn) [inline examples/02_decoding/plot_oasis_vbm.py]:
```suggestion
The data comes from the Open Access Series of Imaging Studies
```

File: examples/02_decoding/plot_oasis_vbm.py [hunk 3]
```diff
@@ -5,228 +5,251 @@
...
 
-# Split data into training set and test set
 from sklearn.model_selection import train_test_split
 
-gm_imgs_train, gm_imgs_test, age_train, age_test = train_test_split(
-    gray_matter_map_filenames, age, train_size=0.6, random_state=0
-)
+from nilearn.datasets import fetch_oasis_vbm
+
+n_subjects = 200  # more subjects requires more memory
```
Comment (hndgzkn) [inline examples/02_decoding/plot_oasis_vbm.py]:
```suggestion
n_subjects = 200  # more subjects require more memory
```

File: examples/02_decoding/plot_oasis_vbm.py [hunk 4]
```diff
@@ -5,228 +5,251 @@
...
+    standardize=None,
+    memory="nilearn_cache",  # cache options
+    verbose=verbose,
+).fit_transform(gray_matter_map_filenames)
+data = variance_threshold.fit_transform(gm_maps_masked)
+
 output = permuted_ols(
     age,
     data,  # + intercept as a covariate by default
     # few permutations in the interest of time; 10000 would be better
```
Comment (hndgzkn) [inline examples/02_decoding/plot_oasis_vbm.py:214]:
```suggestion
    # few permutations in the interest of time
```

**Review by hndgzkn — APPROVED** [review]
LGTM

File: examples/02_decoding/plot_oasis_vbm.py (hunk 4 above)
Comment (hndgzkn) [reply examples/02_decoding/plot_oasis_vbm.py:214]: This appears also near the parameter.

File: examples/02_decoding/plot_oasis_vbm.py [hunk 5]
```diff
@@ -5,228 +5,251 @@
 This example uses Voxel-Based Morphometry (:term:`VBM`)
 to study the relationship between aging and gray matter density.
 
-The data come from the `OASIS <https://sites.wustl.edu/oasisbrains/>`_ project.
+The data come from the Open Access Series of Imaging Studies
+(`OASIS <https://sites.wustl.edu/oasisbrains/>`_) project.
 If you use it, you need to agree with the data usage agreement available
 on the website.
 
 It has been run through a standard :term:`VBM` pipeline (using SPM8 and
```
Comment (hndgzkn) [inline examples/02_decoding/plot_oasis_vbm.py]:
```suggestion
It has been processed through a standard :term:`VBM` pipeline (using SPM8 and
```

Comment (hndgzkn) [reply examples/02_decoding/plot_oasis_vbm.py]: If possible `SPM8` could be written more explicit and given a link, a link for `NewSegment` as well.

File: examples/02_decoding/plot_oasis_vbm.py [hunk 6]
```diff
@@ -5,228 +5,251 @@
...
 
-Note that for an actual predictive modeling study of aging, the study
-should be ran on the full set of subjects. Also, all parameters should be set
-by cross-validation. This includes the smoothing applied to the data and the
-number of features selected by the :term:`ANOVA` step. Indeed, even these
-data-preparation parameter impact significantly the prediction score.
+    For more information
+    see the :ref:`dataset description <oasis_maps>`.
 
+Predictive modeling analysis: VBM bio-markers of aging?
```
Comment (hndgzkn) [inline examples/02_decoding/plot_oasis_vbm.py]: There is a question mark at the end. If this is supposed to be a question, it could be rephrased to be more clear.

File: examples/02_decoding/plot_oasis_vbm.py [hunk 7]
```diff
@@ -5,228 +5,251 @@
...
-much clearer view of the important regions.
+    Note that for an actual predictive modeling study of aging,
+    the study should be ran on the full set of subjects.
 
-.. seealso::
+    Also, all parameters should be set by cross-validation.
+    This includes the smoothing applied to the data and the
+    number of features selected by the :term:`ANOVA` step.
+    Indeed, even these data-preparation parameter
+    impact significantly the prediction score.
```
Comment (hndgzkn) [inline examples/02_decoding/plot_oasis_vbm.py]: I think there is a problem with this sentence.

File: examples/02_decoding/plot_oasis_vbm.py [hunk 8]
```diff
@@ -5,228 +5,251 @@
...
 
-    For more information
-    see the :ref:`dataset description <oasis_maps>`.
+    Also, parameters such as the smoothing should be applied
+    to the data and the number of features selected by the :term:`ANOVA` step
+    should be set by nested cross-validation,
+    as they impact significantly the prediction score.
```
Comment (hndgzkn) [inline examples/02_decoding/plot_oasis_vbm.py]: This paragraph and the above one seems to say the same thing.

File: examples/02_decoding/plot_oasis_vbm.py [hunk 9]
```diff
@@ -5,228 +5,251 @@
...
+#
+# In Nilearn we can benefit from the built-in DecoderRegressor object
+# to do an :term:`ANOVA` with SVR
+# instead of manually defining the whole pipeline.
+#
+# This estimator also uses cross validation to select best models
+# and ensemble them.
+#
+# To save time (because these are anat images with many voxels),
+# we include only the 1-percent voxels most correlated
```
Comment (hndgzkn) [inline examples/02_decoding/plot_oasis_vbm.py]:
```suggestion
# we include only the 1-percent of the voxels most correlated
```

File: examples/02_decoding/plot_oasis_vbm.py [hunk 10]
```diff
@@ -5,228 +5,251 @@
...
 mask = nifti_masker.inverse_transform(variance_threshold.get_support())
 
+
 # %%
-# Prediction pipeline with :term:`ANOVA` and SVR using
-# :class:`~nilearn.decoding.DecoderRegressor` Object
+# Prediction pipeline with ANOVA and SVR using DecoderRegressor
+# -------------------------------------------------------------
+#
+# In Nilearn we can benefit from the built-in DecoderRegressor object
```
Comment (hndgzkn) [inline examples/02_decoding/plot_oasis_vbm.py]:
```suggestion
# In Nilearn we can benefit from the built-in :class:`~nilearn.decoding.DecoderRegressor` object
```

File: examples/00_tutorials/plot_3d_and_4d_niimg.py
Comment (hndgzkn) [inline examples/00_tutorials/plot_3d_and_4d_niimg.py:19]:
```suggestion
print(tmap_filename)

```

Comment (hndgzkn) [inline examples/00_tutorials/plot_3d_and_4d_niimg.py]:
```suggestion
# It is a 4D nifti file. `nilearn.image` package provides some utility functions to work with image files.

# We can load it into the memory to print its shape.

from nilearn.image import load_img

print(load_img(rsn).shape)

```

Comment (hndgzkn) [reply examples/00_tutorials/plot_3d_and_4d_niimg.py]: This is a suggestion to emphasize the functions in image. Using the functions as part of `image` package might be misleading if we are not careful to give the impression that there are image methods. I will make other suggestions in the same sense, but you can ignore them if they do not make sense.

Comment (hndgzkn) [inline examples/00_tutorials/plot_3d_and_4d_niimg.py]:
```suggestion
from nilearn.image import index_img

first_rsn = index_img(rsn, 0)
print(first_rsn.shape)
```

Comment (hndgzkn) [inline examples/00_tutorials/plot_3d_and_4d_niimg.py]:
```suggestion
from nilearn.image import iter_img

for img in iter_img(rsn):
```

Comment (hndgzkn) [inline examples/00_tutorials/plot_3d_and_4d_niimg.py]:
```suggestion
selected_volumes = index_img(rsn, slice(3, 5))
```

Comment (hndgzkn) [inline examples/00_tutorials/plot_3d_and_4d_niimg.py]:
```suggestion
for img in iter_img(selected_volumes):
```

File: examples/00_tutorials/plot_decoding_tutorial.py
Comment (hndgzkn) [inline examples/00_tutorials/plot_decoding_tutorial.py]:
```suggestion
# All we have to do for now is to define a mask filename.
```

Comment (hndgzkn) [inline examples/00_tutorials/plot_decoding_tutorial.py]: This sentence could be moved down. Actually the images are not yet converted to data matrix.

Comment (hndgzkn) [reply examples/00_tutorials/plot_decoding_tutorial.py]: If it is not a very naive question, could we explain why we need to define a mask?

Comment (hndgzkn) [inline examples/00_tutorials/plot_decoding_tutorial.py]: This can be rephrased and put as the first paragraph of this section.

Comment (hndgzkn) [inline examples/00_tutorials/plot_decoding_tutorial.py]:
```suggestion
# The number of the runs is stored in the CSV file giving
```

Comment (hndgzkn) [inline examples/00_tutorials/plot_decoding_tutorial.py]:
```suggestion
# We can plot the weights, using the subject's anatomical image as background.
```

File: examples/00_tutorials/plot_nilearn_101.py
Comment (hndgzkn) [inline examples/00_tutorials/plot_nilearn_101.py]:
```suggestion
from nilearn.plotting import plot_img

plot_img(MNI152_FILE_PATH)
```

Comment (hndgzkn) [inline examples/00_tutorials/plot_nilearn_101.py]:
```suggestion
# on making prettier plots.
```

Comment (hndgzkn) [inline examples/00_tutorials/plot_nilearn_101.py]:
```suggestion
from nilearn.image import smooth_img

smooth_anat_img = smooth_img(MNI152_FILE_PATH, fwhm=3)

```

Comment (hndgzkn) [inline examples/00_tutorials/plot_nilearn_101.py]:
```suggestion
plot_img(smooth_anat_img)

# %%
# We could also pass it to the smoothing function
more_smooth_anat_img = smooth_img(smooth_anat_img, fwhm=3)
plot_img(more_smooth_anat_img)

```

Comment (hndgzkn) [inline examples/00_tutorials/plot_nilearn_101.py]:
```suggestion
anats_all_subjects_smooth = smooth_img(anats_all_subjects, fwhm=5)
```

File: examples/00_tutorials/plot_nilearn_101.py [hunk 11]
```diff
@@ -109,9 +109,6 @@
 plotting.show()
```
Comment (hndgzkn) [inline examples/00_tutorials/plot_nilearn_101.py]:
```suggestion
# Finally, calling `show` function from `nilearn.plotting` package is necessary to display the figure
# when running as a script outside IPython
from nilearn.plotting import show
show()

```

**Review by hndgzkn — APPROVED** [review]
Finally I made a more detailed review (as we talked in the meeting, to look from the perspective of a new user and read the whole example, not just changes). 

I made many suggestions, some are arguable.

File: examples/02_decoding/plot_oasis_vbm.py (hunk 2 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_oasis_vbm.py]: pedantic me: but the singular is "datum", so it should be "the data come"
other me to pedantic me: shut up!

File: examples/02_decoding/plot_oasis_vbm.py (hunk 4 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_oasis_vbm.py:214]: fixed

File: examples/02_decoding/plot_oasis_vbm.py (hunk 6 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_oasis_vbm.py]: shorten the title to avoid the issue

File: examples/02_decoding/plot_oasis_vbm.py (hunk 8 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_oasis_vbm.py]: I hope it reads better now

File: examples/02_decoding/plot_oasis_vbm.py (hunk 2 above)
Comment (hndgzkn) [reply examples/02_decoding/plot_oasis_vbm.py]: Yes, you are right, my mistake :)

Comment (Remi-Gau) [reply examples/02_decoding/plot_oasis_vbm.py]: honnestly I think both are fine

File: examples/00_tutorials/plot_decoding_tutorial.py
Comment (Remi-Gau) [reply examples/00_tutorials/plot_decoding_tutorial.py]: done

Comment (Remi-Gau) [reply examples/00_tutorials/plot_decoding_tutorial.py]: done

Comment (Remi-Gau) [reply examples/00_tutorials/plot_decoding_tutorial.py]: adding an explanation that it restricts which features to use in the decoding
